{"contract": "wallet_03_wrong_constructor", "sourceLanguage": "Solidity", "analysis": {"38": [{"level": "warning", "tool": ["mythril"], "snippet": {"text": "to.transfer(this.balance)"}, "fullDescription": "In the function `migrateTo(address)` a non-zero amount of Ether is sent to an address taken from function arguments.\n\nThere is a check on storage index 0. This storage slot can be written to by calling the function `initWallet()`.", "name": "AccessControlVulnerability"}], "24": [{"level": "note", "tool": ["mythril"], "snippet": {"text": "assert(balances[msg.sender] + msg.value > balances[msg.sender])"}, "fullDescription": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. ", "name": "OtherVulnerability"}, {"level": "warning", "tool": ["oyente"], "name": "ArithmeticVulnerability"}]}, "listLine": [38, 24], "rules": [{"id": "AccessControl_13", "fullDescription": {"text": "In the function `migrateTo(address)` a non-zero amount of Ether is sent to an address taken from function arguments.\n\nThere is a check on storage index 0. This storage slot can be written to by calling the function `initWallet()`."}, "name": "AccessControlVulnerability", "shortDescription": {"text": "Ether send"}}, {"id": "Other_9", "fullDescription": {"text": "A reachable exception (opcode 0xfe) has been detected. This can be caused by type errors, division by zero, out-of-bounds array access, or assert violations. This is acceptable in most situations. Note however that `assert()` should only be used to check invariants. Use `require()` for regular input checking. "}, "name": "OtherVulnerability", "shortDescription": {"text": "Exception state"}}, {"id": "Arithmetic_11", "name": "ArithmeticVulnerability", "shortDescription": {"text": "Integer Overflow."}}]}